lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -438,28 +438,30 @@ def cmd_search(*args)
               cached = true
             end
 
+            if sort_attribute && !valid_sort_attributes.include?(sort_attribute)
+              print_error("Supported options for the -s flag are: #{valid_sort_attributes}")
+              return false
+            end
+
             begin
               if cached
                 print_status('Displaying cached results')
               else
                 search_params = Msf::Modules::Metadata::Search.parse_search_string(match)
                 @module_search_results = Msf::Modules::Metadata::Cache.instance.find(search_params)
 
-                if sort and sort_options.include?(sort)
-                  if sort == 'date'
-                    sort = 'disclosure_date'
-                  end
-                  if sort != 'check'
-                    @module_search_results.sort_by! { |meta| meta.send(sort) }
+                @module_search_results.sort_by! do |module_metadata|
+                  if sort_attribute == 'check'
+                    module_metadata.check ? 0 : 1
+                  elsif sort_attribute == 'disclosure_date' || sort_attribute == 'date'
+                    # Not all modules have disclosure_date, i.e. multi/handler
+                    module_metadata.disclosure_date || Time.utc(0)
                   else
-                    @module_search_results.sort_by! { |meta| meta.send(sort) ? 0 : 1} # Taken from https://stackoverflow.com/questions/14814966/is-it-possible-to-sort-a-list-of-objects-depending-on-the-individual-objects-re
+                    module_metadata.send(sort_attribute)
```

Comment:
Similar comment to above but are we sure all of these modules will always have these functions? If so happy to proceed, just wanted to do a quick double check.

Reply:
Should be good; Context here: https://github.com/rapid7/metasploit-framework/pull/16419#discussion_r844287202 :+1: